c2corg_api/views/report.py

```diff
@@ -56,6 +74,85 @@ def collection_post(self):
     def put(self):
         return self._put(Report, schema_report)
 
+        # if self.request.has_permission('moderator'):
+        #     return self._put(Report, schema_report)
+
+        # # but a normal user can only change a report that he created
+        # elif self._has_permission(self.request.authenticated_userid,
+        #                           self.request.validated['id']):
+        #
+        #     print(self.request.authenticated_userid)
+        #     return self._put(Report, schema_report)
+
+        # else:
+        #     # raise HTTPForbidden('No permission to change this report')
+        #     return self._put(Report, schema_report_without_personal)
+
+    def _has_permission(self, user_id, report_id):
+        """Check if the user with the given id has permission to change the
+        report. That is only users that are currently assigned to the report
+        can modify it.
+        """
```

Comment:
I think this could help:
https://github.com/c2corg/v6_api/blob/master/c2corg_api/views/__init__.py#L182